File: MPP/MPP Proiect/CSharp/persistance/DBRepositories/TestRepository.cs
using System.Collections.Generic;
using System.Data.SQLite;
using log4net;
using model;
using persistance.Interfaces;
using persistance.Utils;

namespace persistance.DBRepositories{
    public class TestRepository: ITestRepository{
        private IDictionary<string, string> _props;

        private static readonly ILog Log = LogManager.GetLogger("TestRepository");

        public TestRepository(IDictionary<string, string> props){
            this._props = props;
            Log.Info("Creating TestRepository");
        }

        public void Add(Test entity){
            Log.Info("TestRepository Add method");
            Log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            using(var command = new SQLiteCommand(
                      "insert into Tests values (@id, @idP, @idR, @points, @date_t)",
                      connection as SQLiteConnection)){
                command.Parameters.AddWithValue("@id", null);
                command.Parameters.AddWithValue("@idP", entity.idParticipant);
                command.Parameters.AddWithValue("@idR", entity.idReferee);
                command.Parameters.AddWithValue("@points", entity.points);
                command.Parameters.AddWithValue("@date_t", entity.date);
                command.ExecuteNonQuery();
                Log.InfoFormat("Succesfully added test {0}", entity);
            }
        }


        public void Remove(Test entity){
            RemoveById(entity.id);
        }

        public void RemoveById(int id){
            Log.Info("TestRepository RemoveByID method");
            Log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            using(var command = new SQLiteCommand("delete from Tests where id = @id",
                      connection as SQLiteConnection)){
                command.Parameters.AddWithValue("@id", id);
                command.ExecuteNonQuery();
                Log.InfoFormat("Succesfully removed test with id {0}", id);
            }
        }

        public List<Test> GetAll(){
            var list = new List<Test>();
            Log.Info("TestRepository GetAll method");
            Log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            using(var cmd = new SQLiteCommand("select * from Tests", connection as SQLiteConnection)){
                using(var reader = cmd.ExecuteReader()){
                    while(reader.Read()){
                        var id = reader.GetInt32(0);
                        var idP = reader.GetInt32(1);
                        var idR = reader.GetInt32(2);
                        var points = reader.GetInt32(3);
                        var date = reader.GetString(4);
                        var test = new Test(idP, idR, points, date);
                        test.id = id;
                        list.Add(test);
                    }
                }
            }

            Log.Info("Successfully created referee's list");
            return list;
        }

        public Test GetById(int id){
            Test test = null;
            Log.Info("TestRepository GetById method");
            Log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            var cmd = new SQLiteCommand("select * from Tests where id = " + id, connection as SQLiteConnection);
            using(var reader = cmd.ExecuteReader()){
                if(reader.Read()){
                    var idP = reader.GetInt32(1);
                    var idR = reader.GetInt32(2);
                    var points = reader.GetInt32(3);
                    var date = reader.GetString(4);
                    test = new Test(idP, idR, points, date);
                    test.id = id;
                }
            }

            Log.Info("Successfully got test");
            return test;
        }
    }
}

File: MPP/MPP Proiect/Java/Persistence/src/main/java/persistence/repository/dbrepositories/TestRepository.java
package persistence.repository.dbrepositories;

import model.Test;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.stereotype.Component;
import persistence.repository.interfaces.ITestRepository;
import persistence.repository.utils.JDBCUtils;

import java.io.Serializable;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Properties;

@Component
public class TestRepository implements ITestRepository, Serializable {

    protected JDBCUtils jdbcUtils;

    private static final Logger logger = LogManager.getLogger();

    public TestRepository() {
        logger.info("Initializing TestRepository with properties: {} ",
                "default");
    }

    public TestRepository(Properties properties) {
        jdbcUtils = new JDBCUtils(properties);
        logger.info("Initializing TestRepository with properties: {} ", properties);
    }

    @Override
    public void add(Test entity) {
        //int idParticipant, int idReferee, int points, String date
        logger.traceEntry("Saving task {}", entity);
        Connection con = jdbcUtils.getConnection();
        try (PreparedStatement preStnt = con
                .prepareStatement("insert into Tests(idParticipant, idReferee, points, date_t)" +
                        "values (?, ?, ?, ?)")) {
            preStnt.setInt(1, entity.getIdParticipant());
            preStnt.setInt(2, entity.getIdReferee());
            preStnt.setInt(3, entity.getPoints());
            preStnt.setString(4, entity.getDate());
            int result = preStnt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit("Successfully saved Test");
    }

    @Override
    public void remove(Test entity) {
        this.removeById(entity.getId());
    }


    @Override
    public void removeById(int id) {
        logger.traceEntry("Saving task {}", id);
        Connection con = jdbcUtils.getConnection();
        try (PreparedStatement preStnt = con
                .prepareStatement("delete from Tests where id = ?")) {
            preStnt.setInt(1, id);
            int result = preStnt.executeUpdate();
            logger.trace("Deleted {} instances", result);
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit("Successfully deleted Test");
    }

    @Override
    public ArrayList<Test> getAll() {
        logger.traceEntry();
        Connection con = jdbcUtils.getConnection();
        ArrayList<Test> tests = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("select * from Tests")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt(1);
                    int idParticipant = result.getInt(2);
                    int idReferee = result.getInt(3);
                    int points = result.getInt(4);
                    String date = result.getString(5);
                    Test test = new Test(idParticipant, idReferee, points, date);
                    test.setId(id);
                    tests.add(test);
                }
            }
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return tests;
    }

    @Override
    public Test getById(Integer id) {
        logger.traceEntry();
        Connection con = jdbcUtils.getConnection();
        Test test = null;
        try (PreparedStatement preStmt = con.prepareStatement("select * from Tests where id = " + id)) {
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    int idParticipant = result.getInt(2);
                    int idReferee = result.getInt(3);
                    int points = result.getInt(4);
                    String date = result.getString(5);
                    test = new Test(idParticipant, idReferee, points, date);
                    test.setId(id);
                }
            }
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return test;
    }

    @Override
    public void update(int id, Test test) {
        logger.traceEntry("Updating test with id {} with {}", id, test);
        Connection con = jdbcUtils.getConnection();
        try (PreparedStatement preStnt = con
                .prepareStatement("update Tests set idParticipant = ?, idReferee = ?, points = ?, date_t = ? where id = ?")) {
            preStnt.setInt(1, test.getIdParticipant());
            preStnt.setInt(2, test.getIdReferee());
            preStnt.setInt(3, test.getPoints());
            preStnt.setString(4, test.getDate());
            preStnt.setInt(5, id);
            int result = preStnt.executeUpdate();
            logger.trace("Updated {} instances", result);
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
    }
}
